### DISCUSSION TITLE
Sometimes it crashes (same condition is true and false at the same time)

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4336] [7. Bugs (archived)] [answered]
Here is an example: `search.active` is both `true` and `false` at the same time. It gets usually triggered by a keyboard event, but that might be just random.

![image](https://github.com/user-attachments/assets/763a7398-6a0c-4cb4-beb5-c63f3ed34993)

**Comment by simonferndriger** [comment]
I can now stedily reproduce it (in my semi-complex app), and the only thing I see in the console are some unrelated warnings.

**Comment by simonferndriger** [comment]
Ah, I could solve it by listening in the body just for `@keyup` instead of `@keyup.window`. Even though I used $event.stopPropagation() it was received 12 times in a very short amount of time, I guess this was just too much stress for AlpineJS

**Comment by ekwoka** [comment, marked as answer]
Hard to say exactly without a recreation, but it could just be related to how the scheduler tries to batch work and prevent circular effects. That they get scheduled and run in a manner that could allow a value to change in between.
